src/main/java/org/opensearch/timeseries/NodeStateManager.java

```diff
@@ -235,68 +157,140 @@ public boolean isMuted(String nodeId, String detectorId) {
     /**
      * When we have a unsuccessful call with a node, increment the backpressure counter.
      * @param nodeId an ES node's ID
-     * @param detectorId Detector ID
+     * @param configId config ID
      */
-    public void addPressure(String nodeId, String detectorId) {
+    public void addPressure(String nodeId, String configId) {
         Map<String, BackPressureRouting> routingMap = backpressureMuter
-            .computeIfAbsent(detectorId, k -> new HashMap<String, BackPressureRouting>());
+            .computeIfAbsent(configId, k -> new HashMap<String, BackPressureRouting>());
         routingMap.computeIfAbsent(nodeId, k -> new BackPressureRouting(k, clock, maxRetryForUnresponsiveNode, mutePeriod)).addPressure();
     }
 
     /**
      * When we have a successful call with a node, clear the backpressure counter.
      * @param nodeId an ES node's ID
-     * @param detectorId Detector ID
+     * @param configId config ID
      */
-    public void resetBackpressureCounter(String nodeId, String detectorId) {
-        Map<String, BackPressureRouting> routingMap = backpressureMuter.get(detectorId);
+    public void resetBackpressureCounter(String nodeId, String configId) {
+        Map<String, BackPressureRouting> routingMap = backpressureMuter.get(configId);
         if (routingMap == null || routingMap.isEmpty()) {
-            backpressureMuter.remove(detectorId);
+            backpressureMuter.remove(configId);
             return;
         }
         routingMap.remove(nodeId);
     }
 
     /**
-     * Check if there is running query on given detector
-     * @param detector Anomaly Detector
-     * @return true if given detector has a running query else false
+     * Get config and execute consumer function.
+     * [Important!] Make sure listener returns in function
+     *
+     * @param configId config id
+     * @param analysisType analysis type
+     * @param function consumer function.
+     * @param listener action listener. Only meant to return failure.
+     * @param <T> action listener response type
      */
-    public boolean hasRunningQuery(AnomalyDetector detector) {
-        return clientUtil.hasRunningQuery(detector);
+    public <T> void getConfig(
```

Comment:
I see we previously didn't have a getDetector here that takes a consumer function as well, what are the main reasons this is added now as an option?

Reply:
This method is defined in https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/task/ADTaskManager.java#L861-L882 and I moved from there (didn't delete it in this PR, will do it in later PRs). I did it because I felt it would be good to put similar getConfig functions in the same class.